### PR TITLE
Local storage implementation

### PR DESCRIPTION
**Pull request by itsmraga-hub**
# 💯 Achievements
- When the user changes the content of any input field, the data is saved to the local storage.
- When the user loads the page if there is any data in the local storage the input fields are pre-filled with this data.
- Learnt about localStorage and sessionStorage.
- Write and read data to and from localStorage.

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -159,3 +159,40 @@ form.addEventListener('submit', (e) => {
...
+  formInputsArr[2].value = localData.message;
+}
+
+// Event listener to submit buton to store the data to local storage before submitting on click
+submitBtn.addEventListener('click', () => {
+  ContactForm.fullName = formInputsArr[0].value;
+  ContactForm.email = formInputsArr[1].value;
+  ContactForm.message = formInputsArr[2].value;
+
+  localStorage.setItem('fullName', ContactForm.fullName);
```
Comment (topeogunleye) [inline main.js]: - [ ] I kindly suggest that you follow the project requirements by ensuring that on page reload and not on submit, the data the user has entered is saved to local storage

- [ ] As written in the project requirements. I kindly suggest you make use of a single object stored to a single local storage registry.
- [ ] I kindly that you turn your data object to a string In order to be saved to the local storage.

## EXAMPLE
`localStorage.setItem('contactFormInput', JSON.stringify(formData));` 

## EXAMPLE
```
submitBtn.addEventListener('input', () => {
const ContactForm  = {
  ContactForm.fullName = formInputsArr[0].value;
  ContactForm.email = formInputsArr[1].value;
  ContactForm.message = formInputsArr[2].value;
}
  localStorage.setItem(''form', ContactForm);
});

```

File: main.js [hunk 2]
```diff
@@ -159,3 +159,40 @@ form.addEventListener('submit', (e) => {
...
+
+// Object to store our collected data
+const ContactForm = {
+  fullName: '',
+  email: '',
+  message: '',
+};
+
+// Check if there is data and if its there, fill the inputs with it
+if (localData.fullName || localData.email || localData.message) {
```
Comment (topeogunleye) [inline main.js]: - you can make use of `windows.onload` listener to save data on page reload
## Example
```
const formInputsArr = Array.from(formInputs);
window.onload = () => {
  const  ContactForm = JSON.parse(localStorage.getItem('form'));

  if (ContactForm) {
    formInputsArr[0].value =ContactForm.fullName;
    formInputsArr[1].value =  ContactForm.email;
    formInputsArr[2].value = ContactForm.message;
  }
};
```

**Review by topeogunleye — CHANGES_REQUESTED** [review]
> # Changes Required 
> ### Hi @itsmraga-hub,
> Nice job, you are almost there.
> 
> 
> 
> ## (Highlights) Good Points: 
> * Good commit messages. :heavy_check_mark:
> * No linter errors. :heavy_check_mark:
> * Followed Github flow :heavy_check_mark:
> * Professional README file. :heavy_check_mark:
> * PR has a good title and summary :heavy_check_mark:
>
> ## (Changes Required) Aspects to improve: 
> * [ ]  Please check my inline comments.
> 
> ## [Optional] suggestions:
> * Nothing to mention.
> 
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> **Cheers and Happy coding! :clap::clap::clap:**
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear
> Please, remember to tag @topeogunleye in your question so I can receive the notification.
>
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Kossi-stack — APPROVED** [review]
## Status: Approved ✔️

**Hi @itsmraga-hub,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**